addons/l10n_ch/models/res_bank.py

```diff
@@ -191,6 +192,10 @@ def _l10n_ch_get_qr_vals(self, amount, currency, debtor_partner, free_communicat
             # _check_for_qr_code_errors ensures we can't have a QR-IBAN without a QR-reference here
             reference_type = 'QRR'
             reference = structured_communication
+        elif self._is_creditor_reference(structured_communication):
```

Comment:
We could remove the `@staticmethod` s and have 

```suggestion
        # The structured communication can also be a valid creditor reference (ISO 11649)
        # reference: https://github.com/arthurdejong/python-stdnum/blob/master/stdnum/iso11649.py
        elif iso11649.is_valid(structured_communication):
            reference_type = 'SCOR'
            reference = iso11649.validate(structured_communication)
```